stablehlo/dialect/TypeInference.cpp

```diff
@@ -2484,12 +2484,19 @@ LogicalResult inferReduceWindowOp(
     return failure();
 
   for (size_t i = 0; i < inputArgTypes.size(); ++i) {
-    if (!inputArgTypes[i].hasRank())
+    if (!inputArgTypes[i].hasRank()) {
       inferredReturnShapes.emplace_back(inputArgTypes[i].getElementType());
-    else
-      inferredReturnShapes.emplace_back(
-          inferWindowOutputShape(inputArgTypes[i].getShape(), inferredWindow),
-          inputArgTypes[i].getElementType());
+    } else {
+      auto inputRankedType = inputs[i].getType().cast<RankedTensorType>();
+      auto resultShape =
+          inferWindowOutputShape(inputArgTypes[i].getShape(), inferredWindow);
+      encodingToBounds(inputRankedType.getEncoding()).empty()
+          ? inferredReturnShapes.emplace_back(resultShape,
+                                              inputArgTypes[i].getElementType())
+          : inferredReturnShapes.emplace_back(resultShape,
+                                              inputArgTypes[i].getElementType(),
+                                              inputRankedType.getEncoding());
```

Comment:
here, we need to infer the bounds as well depending on the strides, window size. Let me know if you need more inputs on this.